drivers/hmis/app/graphql/types/hmis_schema/referral_posting_input.rb

```diff
@@ -10,9 +10,15 @@ class HmisSchema::ReferralPostingInput < BaseInputObject
     argument :status, ID, required: false
     argument :denial_reason, ID, required: false
     argument :denial_note, String, required: false
+    argument :referral_result, ID, required: false
```

Comment:
Typically we would use enums to constrain the input a but more. Would it make sense to accept `Types::HmisSchema::Enums::Hud::ReferralResult` (which has the numeric values attached to it)? I think if we did that, you don't need to do the `downcase` thing below which seems to be relying on the fact that the GraphQL Enum values and the Active Record Enum values match. Is that right?

This is a bit different from usual since we don't typically use AR enums on the models.

It's also an odd case with the picklist, too – usually we would just use the `Types::HmisSchema::Enums::Hud::ReferralResult` enum directly to generate a pick list (simply `"pickListReference": "ReferralResult"`), but we can't do that in this case because we need to omit one of the values. So I think what you did in PickListOption is alright, though it feels a bit off that it's hard-coding the string values as opposed to referencing the Enum class. Though, I'm not sure if doing `Types::HmisSchema::Enums::Hud::ReferralResult.key_for(2)` is better/clearer.

Reply:
Thanks, using enums seems like a better approach. I'm still learning the mechanics of our dynamic forms.

The picklist options are bit awkward here but I think `Types::HmisSchema::Enums::Hud::ReferralResult.key_for(2)` is better since it throws an error if the key isn't defined